internal/collector/postgres_metrics.go

```diff
@@ -182,29 +183,60 @@ func EnablePostgresMetrics(ctx context.Context, inCluster *v1beta1.PostgresClust
 
 		// Add custom queries if they are defined in the spec
 		if inCluster.Spec.Instrumentation != nil &&
-			inCluster.Spec.Instrumentation.Metrics != nil &&
-			inCluster.Spec.Instrumentation.Metrics.CustomQueries != nil &&
-			inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add != nil {
-
-			for _, querySet := range inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add {
-				// Create a receiver for the query set
-				receiverName := "sqlquery/" + querySet.Name
-				config.Receivers[receiverName] = map[string]any{
-					"driver": "postgres",
-					"datasource": fmt.Sprintf(
-						`host=localhost dbname=postgres port=5432 user=%s password=${env:PGPASSWORD}`,
-						MonitoringUser),
-					"collection_interval": querySet.CollectionInterval,
-					// Give Postgres time to finish setup.
-					"initial_delay": "15s",
-					"queries": "${file:/etc/otel-collector/" +
-						querySet.Name + "/" + querySet.Queries.Key + "}",
+			inCluster.Spec.Instrumentation.Metrics != nil {
+
+			if inCluster.Spec.Instrumentation.Metrics.CustomQueries != nil &&
+				inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add != nil {
+
+				for _, querySet := range inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add {
+					// Create a receiver for the query set
+
+					db := "postgres"
+					if querySet.Database != "" {
+						db = querySet.Database
+					}
+					receiverName := "sqlquery/" + querySet.Name
+					config.Receivers[receiverName] = map[string]any{
+						"driver": "postgres",
+						"datasource": fmt.Sprintf(
+							`host=localhost dbname=%s port=5432 user=%s password=${env:PGPASSWORD}`,
+							db,
+							MonitoringUser),
```

Comment:
:question: Will the monitoring user always have access to whatever databases the user specifies?

Reply:
Looking into this -- I got the metrics I expected, but why?

Reply:
It might be because we `GRANT pg_monitor to ccp_monitoring;` in the setup.